sourcefiles/logicfactory.py

```diff
@@ -1784,3 +1805,43 @@ def getGameConfig(settings: rset.Settings, config: cfg.RandoConfig):
 
     return CfgType(settings, config)
 # end getGameConfig
+
+
+def get_rocksanity_location_groups():
+    denadoroRock = LocationGroup(
+        "Denadoro Rock", 1, lambda game: True, lambda weight: weight-1
```

Comment:
Not sure exactly what "weight" does. Copied from logic for adding Denadoro chests from Chronosanity.

Reply:
The logic code that we used was originally developed for Chronosanity, where KIs are assigned to groups based on the group's weight.  For standard logic (or modes derived from standard logic like this) it's correct to have weight equal to the number of checks.  I think the filler for the standard logic ignores group weights, but a weight of 1 is correct.

Reply:
I think it's worth considering whether you want the rocks to be in their own groups or to be appended to the existing groups for Giant's Claw and Denadoro.

Reply:
Thank you for the feedback!

In trying to understand a bit more about the codebase, can you explain the difference between being in their own location groups or existing ones, a bit more (or link to doc/code about it, if have that)? Is it something about weighting so that a particular "group" is not full of a bunch of KI?

So right now, they are defined as separate groups here, so that means they get more heavily(?) weighted (and more likely to have an item?)?

Alternatively, the Denadoro and Giant's Claw rocks could be added to the existing groups for each of those, so they're just one more check in those locations? Happy to implement that if you think that makes more sense.

Reply:
Having them in separate groups can give you more granular control over weighting and how often you want that spot to be selected. Weighting for most groups drops off sharply after a KI has been assigned, so it's much less likely to get another.  

We did quite a bit of fiddling with existing groups and weighting back in the day to get a reasonable item distribution.  I'd have to resurrect the analysis scripts, but we're probably due for another round of balancing with the new locations and items.